qute.ls/com.redhat.qute.ls/src/test/java/com/redhat/qute/services/diagnostics/QuteDiagnosticsInExpressionWithWithSectionTest.java

```diff
@@ -34,23 +34,28 @@ public void undefinedObject() throws Exception {
 		String template = "{#with item}\r\n" + //
 				"{/with}";
 
-		Diagnostic d = d(0, 7, 0, 11, QuteErrorCode.UndefinedVariable, "`item` cannot be resolved to a variable.",
+		Diagnostic d1 = d(0, 7, 0, 11, QuteErrorCode.UndefinedVariable, "`item` cannot be resolved to a variable.",
 				DiagnosticSeverity.Warning);
-		d.setData(DiagnosticDataFactory.createUndefinedVariableData("item", false));
+		d1.setData(DiagnosticDataFactory.createUndefinedVariableData("item", false));
 
-		testDiagnosticsFor(template, d);
-		testCodeActionsFor(template, d, //
-				ca(d, te(0, 0, 0, 0, "{@java.lang.String item}\r\n")));
+		Diagnostic d2 = d(0, 0, 1, 7, QuteErrorCode.NotRecommendedWithSection,
+				"`with` is not recommended. Use `let` instead.", DiagnosticSeverity.Warning);
+
+		testDiagnosticsFor(template, d1, d2);
+		testCodeActionsFor(template, d1, //
+				ca(d1, te(0, 0, 0, 0, "{@java.lang.String item}\r\n")));
 	}
 
 	@Test
-	public void noError() throws Exception {
+	public void singleSection() throws Exception {
 		String template = "{@org.acme.Item item}\r\n" + //
 				"{#with item}\r\n" + //
 				"  <h1>{name}</h1>  \r\n" + //
 				"  <p>{price}</p> \r\n" + //
 				"{/with}";
-		testDiagnosticsFor(template);
+		Diagnostic d = d(1, 0, 4, 7, QuteErrorCode.NotRecommendedWithSection,
+				"`with` is not recommended. Use `let` instead.", DiagnosticSeverity.Warning);
+		testDiagnosticsFor(template, d);

```

Comment:
Once you will manage code action with severaltext edit,please write test with your code action.

Reply:
Is this the appropriate code action: https://github.com/redhat-developer/quarkus-ls/pull/563/files#diff-f58ddd1bbb8c91147063f45a0973b0babf402ec1873f66899f5e89f4aa3d306bR93-R96 ?

Reply:
perfect!

Reply:
Add a test with #with without object parameter.

Reply:
If there isn't an object parameter, there is simply just a syntax error correct?

Reply:
There is no code action.